### PR TITLE
Align nuxt server url with axios base url

### PR DESCRIPTION
**Pull request by richard-cox**
- This PR fixes the issue i've seen on my linux box when using a localhost and non-localhost `API` env, though I couldn't fully test the later due to my rancher docker container failing to start with a default local cluster
- I still see a `422 (Unprocessable Entity)` (Invalid CSRF token) error on log in when the 'CSRF' cookie value has not been set (fresh log in after clearing cookies, trying to log in again after log out, etc). Confirming if this happens on an otherwise fine system without this change would be worthwhile.
- The crux of the issue seems to be that axios's baseUrl was defaulting to the machine's hostname leading, interestingly, to `Loading error getaddrinfo ENOTFOUND <machine's host name>` on load of landing page
- This can be fixed by updating the local hosts file... but that leads to...
  - `ERR_CERT_AUTHORITY_INVALID` errors due to nuxt's proxy setting the  `x-api-host` to hostname
  - Then also CORS issues when app goes to the `API` urls directly (and not proxied via nuxt)
- This PR addresses the core issue, rather than changes in https://github.com/richard-cox/dashboard/pull/1 which work around it

### CODE REVIEW

**Comment by vincent99** [comment]
For the CSRF thing, seeing an error when it's not set is correct, but it should have been set.  This is a problem with starting from scratch + server-side-rendering, the cookie is sent from rancher->nuxt server but not retransmitted from nuxt->client so the browser never gets it.  This only happens in dev since we don't use currently any of the SSR in real releases.  Addressed in https://github.com/vincent99/dashboard/commit/9980eecf303ac2c7d2d8d7e814b957ba6e9543a9 as part of auth changes.

**Comment by vincent99** [comment]
TL;DR: I think `x-api-host` just needs to be set right in this case, which I've done in https://github.com/rancher/dashboard/pull/2139/files .  Combined with the previous CORS fix should address your problems.

I think there's some confusion about what's supposed to happen here, plus 2 different issues that depend on the combination of several variables:
  - Build mode (dev vs prod)
  - SSR mode (SSR vs SPA)
  - The URL you're hitting to get to nuxt
  - The API env var for where Rancher is

API responses from the server return lots of links to related resources, and the UI follows these to make subsequent requests.  These are always _absolute_ URLs (i.e. starting with `https://`). When Rancher is the one serving up the UI (as always in production), there's no special behavior needed here, because the origin already matches the API.

For dev and SSR, the request get proxied through the nodejs server: 

`Browser  ––localhost:8005 request––> nuxt server ––rancher-server:443 request ––> Rancher API`

To produce responses with links the browser can follow, the API needs to be told to that links should be generated with a base URL that match the _original_ browser request (typically `https://localhost:8005`), not the actual host header the server received (e.g. `https://rancher-server`).  This is done by setting the `x-api-host` header in the proxied request.

If this _isn't_ done right then the UI ends up getting back an API response with links in it that don't match its window.origin, and requests to follow it are blocked by the browser because they are cross-origin.  (Or in your case, they're for a name that the browser can't even resolve to try a preflight request)

The `host` option to nuxt controls what interface the node server is listening on; the default is localhost, so setting it to `0.0.0.0` makes it available on all interfaces of your laptop (/wherever `yarn dev` is running).  This was done intentionally so that e.g. you can look at your running dev server from your phone or see someone els's dev inside of the local network.

So there's just a case where for isDev + SSR-rendered requests the `x-api-host` is set wrong, and your change to set `host` indirectly eliminates a path to that, but also restricts how you can reach the dev server.

**Comment by richard-cox** [comment]
Awesome, thanks for the tweaks and detailed explanation. Tested on master on my non-local API env var works fine.
